lib/experimental/modules/class-gutenberg-modules.php

```diff
@@ -20,64 +20,90 @@ class Gutenberg_Modules {
-				'dynamic' => isset( $dependencies['dynamic'] ) ? $dependencies['dynamic'] : array(),
-			);
+			$deps = array();
+			foreach ( $dependencies as $dependency ) {
+				if ( is_array( $dependency ) && isset( $dependency['id'] ) ) {
+					$deps[] = array(
+						'id'   => $dependency['id'],
+						'type' => isset( $dependency['type'] ) && 'dynamic' === $dependency['type'] ? 'dynamic' : 'static',
+					);
+				} elseif ( is_string( $dependency ) ) {
+					$deps[] = array(
+						'id'   => $dependency,
+						'type' => 'static',
+					);
+				}
+			}
 
 			self::$registered[ $module_identifier ] = array(
 				'src'          => $src,
 				'version'      => $version,
+				'enqueued'     => in_array( $module_identifier, self::$enqueued_modules_before_register, true ),
 				'dependencies' => $deps,
 			);
 		}
 	}
 
 	/**
-	 * Enqueues a module in the page.
+	 * Marks the module to be enqueued in the page.
 	 *
 	 * @param string $module_identifier The identifier of the module.
 	 */
 	public static function enqueue( $module_identifier ) {
-		// Add the module to the queue if it's not already there.
-		if ( ! in_array( $module_identifier, self::$enqueued, true ) ) {
-			self::$enqueued[] = $module_identifier;
+		if ( isset( self::$registered[ $module_identifier ] ) ) {
+			self::$registered[ $module_identifier ]['enqueued'] = true;
+		} elseif ( ! in_array( $module_identifier, self::$enqueued_modules_before_register, true ) ) {
+			self::$enqueued_modules_before_register[] = $module_identifier;
+		}
+	}
+
+	/**
+	 * Unmarks the module so it is not longer enqueued in the page.
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public static function dequeue( $module_identifier ) {
+		if ( isset( self::$registered[ $module_identifier ] ) ) {
+			self::$registered[ $module_identifier ]['enqueued'] = false;
+		}
+		$key = array_search( $module_identifier, self::$enqueued_modules_before_register, true );
+		if ( false !== $key ) {
+			array_splice( self::$enqueued_modules_before_register, $key, 1 );
 		}
 	}
 
 	/**
 	 * Returns the import map array.
 	 *
-	 * @return array Associative array with 'imports' key mapping to an array of module identifiers and their respective source strings.
+	 * @return array Array with an 'imports' key mapping to an array of module identifiers and their respective source URLs, including the version query.
 	 */
 	public static function get_import_map() {
 		$imports = array();
-		foreach ( self::get_dependencies( self::$enqueued, array( 'static', 'dynamic' ) ) as $module_identifier => $module ) {
+		foreach ( self::get_dependencies( array_keys( self::get_enqueued() ) ) as $module_identifier => $module ) {
 			$imports[ $module_identifier ] = $module['src'] . self::get_version_query_string( $module['version'] );
 		}
 		return array( 'imports' => $imports );
 	}
 
 	/**
-	 * Prints the import map.
+	 * Prints the import map using <script type="importmap">.
```

Comment:
I have started finding some errors in IDEs with real HTML tags lately. They seem to be alleviated by surrounding real tags with the backtick.

```php
 * Prints the import map using `<script type="importmap">`.
```

We might also want to consider describing the tag as I think there's precedent for [avoiding actual HTML tags in the documentation](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#formatting-guidelines).

Reply:
Makes sense. Done in https://github.com/WordPress/gutenberg/pull/57231/commits/5120b7d213304799b611addb94adf6389d80c362.